dhall/src/Dhall/TypeCheck.hs

```diff
@@ -3037,7 +3098,11 @@ prettyTypeMessage (UnusedHandler ks) = ErrorMessages {..}
 
 prettyTypeMessage (MissingHandler ks) = ErrorMessages {..}
   where
-    short = "Missing handler"
+    short = case Data.Set.toList ks of
+         []       -> "Missing handler"
```

Comment:
Maybe we should use `NonEmpty` instead – might be overkill though…

Reply:
It does not communicate the invariant that elements are unique and doing both seems.. well too much for just an error message. But I'll think about this.

Reply:
Oh right – I missed that it's a `Set`.

Reply:
Well.. I went and fix that anyway..

Reply:
Hmm, I'm not sure – it's a bit unusual and also inconsistent with what we do for `UnusedHandler`. @Gabriel439, what do you think?

Apart from this – LGTM! :)

Reply:
Yeah.. I'm neutral about this. Anyway, I forgot to fix the long message, meaning that there is yet another commit about this. Just in case you want to keep it.